pkg/api/serialization_test.go

```diff
@@ -247,12 +251,15 @@ func TestUnversionedTypes(t *testing.T) {
 			continue
 		}
 
+		versionDecodedObject := pair.EmptyCopy
 		// Make sure the versioned codec under test can decode
-		versionDecodedObject, err := runtime.Decode(testapi.Default.Codec(), unversionedJSON)
+		err = runtime.DecodeInto(testapi.Default.Codec(), unversionedJSON, versionDecodedObject)
 		if err != nil {
 			t.Errorf("%v: unexpected error: %v", obj, err)
 			continue
 		}
+		// DecodeInto will set the APIVersion to the version the object are registered with, which we don't care.
+		versionDecodedObject.GetObjectKind().SetGroupVersionKind(&unversioned.GroupVersionKind{})
```

Comment:
Technically anything unversioned is not converted, and we don't clear the values automatically if you aren't converted.

I'm on the fence on whether we should clear those values - if it has no impact internally (because we don't read them) I'm disinclined to make other changes.